app/services/service_result.rb

```diff
@@ -162,21 +205,45 @@ def add_dependent!(dependent)
     self.dependent_results += inner_results
   end
 
+  ##
+  # Callback to be executed if the Service call succeeds.
```

Comment:
Tense of the sentence, as well as the word "Callback" imply that `ServiceResult` is some kind of promise, where the result is yet unknown, but at some point in the future the result will be known and then the callback will be executed.

But actually these methods are intended to replace `if` based control-flow statements.

My take:

```suggestion
  # Execute block argument if the Service call succeeded.
```

Similarly I'd also change the docs for `on_failure` and other following methods.

Reply:
You're right - "callback" implies asynchronous behaviour. Will correct both methods.